docs/src/main/asciidoc/pubsub.adoc

```diff
@@ -219,6 +219,10 @@ h|pullNext | Allows for a single message to be pulled and automatically acknowle
 h|pullAndConvert | Works the same as the `pull` method and, additionally, converts the Pub/Sub binary payload to an object of the desired type, using the converter configured in the template.
 |===
 
+WARNING: Setting `returnImmediately` to `true` is not recommended as it may result in long delays in message delivery.
```

Comment:
I'd prefer more active voice:

- We recommend setting `returnImmediately` to `true`, otherwise it may result in delayed message delivery.

- The implementation really means 1 second, and if the client cannot retrieve any messages in that time from the Pub/Sub backend, it will not retrieve any message at all (may need to clarify this sentence?)

 - We recommend setting `returningImmediately` to false, or use one of the `subscribe` methods from the previous section.

Reply:
Thanks! How about now?

Reply:
Much better, though you may want to say "if the Pub/Sub _client_ cannot  get messages from the backend ... " because to me, Pub/Sub is the _service_ so it doesn't make sense that the service can't get messages from itself.

Reply:
Based on the conversation with the Pub/Sub team, it seems more like the latter because of the distributed nature of Pub/Sub. So, omission of the word "client" was intentional.